src/clj/rems/workflow/dynamic.clj

```diff
@@ -827,6 +908,56 @@
                               [{:type ::invite-member :actor "assistant" :member {:name "Member Applicant 1" :email "member1@applicants.com"}}]
                               injections)))))))
 
+(deftest test-remove-member
+  (let [application {:state ::submitted
+                     :members [{:userid "applicant"} {:userid "somebody"}]
+                     :applicantuserid "applicant"
+                     :workflow {:type :workflow/dynamic
+                                :handlers ["assistant"]}}
+        injections {}]
+    (testing "remove member by applicant"
+      (is (= [{:userid "applicant"}]
+             (:members
+              (apply-commands application
+                              [{:type ::remove-member :actor "applicant" :member {:userid "somebody"}}]
+                              injections)))))
+    (testing "remove member by handler"
+      (is (= [{:userid "applicant"}]
+             (:members
+              (apply-commands application
+                              [{:type ::remove-member :actor "assistant" :member {:userid "somebody"}}]
+                              injections)))))
+    (testing "only members can be removed"
+      (is (= {:errors [{:type :user-not-member :user {:userid "notamember"}}]}
+             (handle-command {:type ::remove-member :actor "assistant" :member {:userid "notamember"}}
+                             application
+                             injections))))))
```

Comment:
I tried adding a test that tries to remove applicant by applicant. It worked – not sure if it should :thinking:

Reply:
It shouldn't work anymore